support/cas-server-support-oauth-core-api/src/main/java/org/apereo/cas/support/oauth/web/endpoints/OAuth20RevocationEndpointController.java

```diff
@@ -0,0 +1,194 @@
+                .build();
+
+            val accessResult = getOAuthConfigurationContext().getRegisteredServiceAccessStrategyEnforcer().execute(audit);
+            if (accessResult.isExecutionFailure()) {
+                return OAuth20Utils.writeError(response, OAuth20Constants.INVALID_REQUEST);
+            }
+        }
+        val token = context.getRequestParameter(OAuth20Constants.TOKEN)
+            .map(String::valueOf).orElse(StringUtils.EMPTY);
+
+        return generateRevocationResponse(token, clientId, response);
+    }
+
+    /**
+     * Generate revocation token response.
+     *
+     * @param token the token to revoke
+     * @param clientId the client who requests the revocation
+     * @param response the response
+     * @return the model and view
+     */
+    protected ModelAndView generateRevocationResponse(final String token,
+                                                      final String clientId,
+                                                      final HttpServletResponse response) {
+
+        val registryToken = getOAuthConfigurationContext().getTicketRegistry().getTicket(token, OAuth20Token.class);
+
+        if (registryToken == null) {
+            LOGGER.error("Provided token [{}] has not been found in the ticket registry", token);
+        } else if (isRefreshToken(registryToken) || isAccessToken(registryToken)) {
+            if (!StringUtils.equals(clientId, registryToken.getClientId())) {
+                LOGGER.error("Provided token [{}] has not been issued for the service [{}]", token, clientId);
+                return OAuth20Utils.writeError(response, OAuth20Constants.INVALID_REQUEST);
+            }
+
+            if (isRefreshToken(registryToken)) {
+                revokeToken((OAuth20RefreshToken) registryToken);
+            } else {
+                revokeToken(registryToken.getId());
+            }
+        } else {
+            LOGGER.error("Provided token [{}] is either not a refresh token or not an access token", token);
+            return OAuth20Utils.writeError(response, OAuth20Constants.INVALID_REQUEST);
+        }
+
+        val mv = new ModelAndView(new MappingJackson2JsonView());
+        mv.setStatus(HttpStatus.OK);
+        return mv;
+    }
+
+    /**
+     * Revoke the provided Refresh Token and it's related Access Tokens.
+     *
+     * @param token the token
+     * @return the model and view
+     */
+    private void revokeToken(final OAuth20RefreshToken token) {
+        revokeToken(token.getId());
+
+        token.getAccessTokens().forEach(item-> {
+            revokeToken(item);
+        });
+    }
+
+    /**
+     * Revoke the provided OAuth Token.
+     *
+     * @param token the token
+     * @return the model and view
+     */
+    private void revokeToken(final String token) {
+        LOGGER.debug("Revoking token [{}]", token);
+        getOAuthConfigurationContext().getTicketRegistry().deleteTicket(token);
+    }
```

Comment:
I think doing calls directly specially with the JPA ticket registry fails with a transaction-required error. Do you think you might be able to verify this?

Reply:
I didn't tested it with JPA (only with Redis) but it's exactly the same revocation process than the old one.

Will test it to be sure.

Reply:
OK, thanks. Let's leave it aside for now. I can try to help in this area too.